apps/src/lib/node/ledger/storage/rocksdb.rs

```diff
@@ -273,32 +281,44 @@ impl RocksDB {
         println!("Will write to {} ...", full_path.to_string_lossy());
 
         let mut dump_it = |prefix: String| {
-            for next in self.0.iterator(IteratorMode::From(
-                prefix.as_bytes(),
-                Direction::Forward,
-            )) {
-                match next {
-                    Err(e) => {
-                        eprintln!(
-                            "Something failed in a \"{prefix}\" iterator: {e}"
-                        )
-                    }
-                    Ok((raw_key, raw_val)) => {
-                        let key = std::str::from_utf8(&raw_key)
-                            .expect("All keys should be valid UTF-8 strings");
-                        let val = HEXLOWER.encode(&raw_val);
-                        let bytes = format!("\"{key}\" = \"{val}\"\n");
-                        file.write_all(bytes.as_bytes())
-                            .expect("Unable to write to output file");
-                    }
-                };
+            let mut read_opts = ReadOptions::default();
+            read_opts.set_total_order_seek(true);
+
+            let mut upper_prefix = prefix.clone().into_bytes();
+            if let Some(last) = upper_prefix.pop() {
+                upper_prefix.push(last + 1);
+            }
+            read_opts.set_iterate_upper_bound(upper_prefix);
+
+            let iter = self.0.iterator_opt(
+                IteratorMode::From(prefix.as_bytes(), Direction::Forward),
+                read_opts,
+            );
+
+            for (key, raw_val, _gas) in PersistentPrefixIterator(
+                PrefixIterator::new(iter, String::default()), /* Empty string to prevent prefix stripping, the prefix is already in the enclosed iterator */
+            ) {
+                let val = HEXLOWER.encode(&raw_val);
+                let bytes = format!("\"{key}\" = \"{val}\"\n");
+                file.write_all(bytes.as_bytes())
+                    .expect("Unable to write to output file");
             }
         };
 
-        // Dump accounts subspace and block height data
         dump_it("subspace".to_string());
-        let block_prefix = format!("{}/", height.raw());
-        dump_it(block_prefix);
+
+        if extended_dump {
+            // Dump the keys prepended with the selected block height (includes
+            // subspace diff keys)
+            dump_it(height.raw());
```

Comment:
In here I manually specify the prefixes to dump. This is a bit tedious and also not future-proof since we'd need to update this list in case we changed the well-known keys that we write to storage.

An alternative to this could be to iterate over all the keys given the prefix `"a"`: given the lexicographical order that Namada imposes on RocksDB this prefix should include all the keys but the height-prepended one. I didn't choose this solution because of two doubts that I still have:

- What happens if at a certain point the encoded value of a height starts with an `"a"`? I guess we would have duplicated keys in the dump which is what we already suffered from while debugging the testnet
- We'd still need to manage separately the `results` subkey, since this one is **followed** by the height (is there a reason why we do this and not te opposite as for the other keys?)